branchff

```diff
@@ -92,13 +90,19 @@ gitlib::repo_state || common::exit 1
 ###############################################################################
 # MAIN
 ###############################################################################
-BASEDIR="/usr/local/google/$USER"
+BASEDIR="$(mktemp -d "$TMPDIR/$PROG.XXXXXX")"
 # WORK/BUILD area
 WORKDIR=$BASEDIR/$PROG-$RELEASE_BRANCH
 # Go tools expect the kubernetes src to be under $GOPATH
 export GOPATH=$WORKDIR
 # TREE_ROOT is working branch/tree
 TREE_ROOT=$WORKDIR/src/k8s.io/kubernetes
+
+# Set a trap to cleanup BASEDIR if --cleanup is set.
+if (($FLAGS_cleanup)) ; then
+  trap 'rm -rf "$BASEDIR"' EXIT ERR SIGINT SIGQUIT SIGTERM SIGHUP
+fi
```

Comment:
Nit:
instead of checking twice for the flag, we could just have a cleanup function which is called in the trap, something like:

```sh
function cleanup() {
  if (($FLAGS_cleanup)) ; then
    logecho "$BASEDIR has been removed, as CLEANUP environment variable was set."
    rm -rf "$BASEDIR"
  else
    logecho "$WORKDIR left intact for reviewing"
  fi
}

trap cleanup EXIT
```

Reply:
@hoegaarden -- I considered a cleanup function, but decided against it because of that `else`. I only want `WORKDIR` to be left intact if we:
- reach the end of the script, successfully
- `--cleanup` is not set

The way it runs currently, it will cleanup if `--cleanup` is set AND the script exits for various reasons e.g., `^C`